conans/client/hook_manager.py

```diff
@@ -99,7 +100,9 @@ def _load_module_from_file(hook_path):
             old_modules = list(sys.modules.keys())
             with chdir(current_dir):
                 sys.dont_write_bytecode = True
-                loaded = __import__(filename)
+                import importlib
+                importlib.invalidate_caches()
```

Comment:
This is failing in CI:

> AttributeError: 'module' object has no attribute 'invalidate_caches'

It doesn't exist in py27. Even if we are deprecating it, we would need to keep it working for a few months, to give time to transition. Furthermore the ``invalidate_caches()`` seems more related to the time of creation of the files, in this case, files are there before starting the interpreter.

Reply:
The `invalidate_caches` araised because before it the CI failed in the `ConanInstallTest.reuse_python_test`. It couldn't find the module in the `python` path _only_ if we tried to manually import a hook before the execution of a command. 

I'm not fully sure since I'm a bit out of my depth on the Python knowledge here but I thought that it was the case for which the `invalidate_caches` was introduced since we dynamically modify the `sys.path` in both `conan_api` and `hook_manager`. As per documentation:
> function should be called if any modules are created/installed while your program is running

Reply:
Okay, it seems that my "fix" was superficial since it fails again on the 3.6 version with the module not found in the `reuse_python_test`. I'm quite puzzled. 
My only guess is that the problem occurs only in tests due to consequential `client.run` calls where each one creates an API instance which modifies the `sys.path`. On the first call, everything is working and found. On the next call, it sees the cached modules and fails to load them. Because we don't clean up properly on the API destruction?

Reply:
I've resolved the problem. The `invalidate_caches` contains a bug in the Python 3.6 and less to not properly clean up lost finders. See `https://bugs.python.org/issue33169`. 
The situation is very tricky. As I understand from the issue description is that If we do multiple commands in the same interpreter run, tests or a Python API for example, there is a situation where if we don't import modules the first time after adding a path, but do try to do that the second time - the cached finder for that path in the `sys.path_importer_cache` will be `None`. I could reproduce it locally, weird stuff.

After backporting the implementation from 3.7 and doing a full cache clean up in 2.7 everything is passing.